outport/process/outportDataProvider.go

```diff
@@ -120,6 +124,101 @@ func (odp *outportDataProvider) PrepareOutportSaveBlockData(arg ArgPrepareOutpor
 	}, nil
 }
 
+func collectExecutedTxHashes(bodyHandler data.BodyHandler, headerHandler data.HeaderHandler) (map[string]struct{}, error) {
+	executedTxHashes := make(map[string]struct{})
+	mbHeaders := headerHandler.GetMiniBlockHeaderHandlers()
+	body, ok := bodyHandler.(*block.Body)
+	if !ok {
+		return nil, ErrWrongTypeAssertion
+	}
+
+	miniBlocks := body.GetMiniBlocks()
+	if len(miniBlocks) != len(mbHeaders) {
+		return nil, ErrMiniBlocksHeadersMismatch
+	}
+
+	for i, mbHeader := range mbHeaders {
+		if mbHeader.GetTypeInt32() == int32(block.PeerBlock) {
+			continue
+		}
+		if mbHeader.GetConstructionState() == int32(block.Processed) {
+			continue
+		}
+
+		for j := mbHeader.GetIndexOfFirstTxProcessed(); j <= mbHeader.GetIndexOfLastTxProcessed(); j++ {
+			txHash := miniBlocks[i].TxHashes[j]
+			executedTxHashes[string(txHash)] = struct{}{}
+		}
+	}
+
+	return executedTxHashes, nil
+}
+
+func (odp *outportDataProvider) setExecutionOrderInTransactionPoolWithChecks(
+	pool *outportcore.Pool,
+	executedTxHashes map[string]struct{},
+) {
+	orderedTxHashes := odp.executionOrderHandler.GetItems()
+	txGroups := []map[string]data.TransactionHandlerWithGasUsedAndFee{
+		pool.Txs,
+		pool.Scrs,
+		pool.Receipts,
+		pool.Rewards,
+	}
+
+	foundTxHashes := 0
+	for i, txHash := range orderedTxHashes {
+		for _, group := range txGroups {
+			if setExecutionOrderIfFound(txHash, group, i) {
+				foundTxHashes++
+				break
+			}
+		}
+	}
+
+	err := checkTxOrder(orderedTxHashes, executedTxHashes, foundTxHashes)
```

Comment:
optional: could have been extracted in a new check function and this function could have just been related to the setting of the execution the other one just checked that everything is ok

Reply:
done